### DISCUSSION TITLE
8 July 2024 1355

### DISCUSSION THREAD

**Discussion by ktoddbrown** [ktoddbrown/SoilDRaH#26] [RegularCalls]
### (5 | 1400) Check ins

- Who is here
- Note taking?
- How are you showing up? [Feels wheel](https://feelingswheel.com/)

### (45 | 1405) Issues checkins and stuck to unstuck
Pull up open issues and walk through in reverse order.

### (5 | 1455) Check outs [appreciations, curiosities, and ToDo]

**Comment by graham-louis** [comment]
https://github.com/ktoddbrown/SoilDRaH/discussions/26 

### (5 | 1400) Check ins 

- Who is here 
    - Vaasuki, Graham, Gabriel, Ashley Lang, Kate Heckman, Kathe
- Note taking? - Graham
- How are you showing up? [Feels wheel](https://feelingswheel.com/)

### (45 | 1405) Issues checkins and stuck to unstuck

- CPEAT [https://github.com/ktoddbrown/SoilDRaH/issues/1](https://github.com/ktoddbrown/SoilDRaH/issues/1)
    - Updated model file for CPEAT report; merge branch?
    - Unable to check using checkAnnotations
    - Column names successfully synchronized with annotations
    - Would like some help with structure of the integrated data product
    - Will add check boxes to the issue and tag Dr. TB
    - HTML (Comparative figure) was shared but still needs review
- ISCN3 [https://github.com/ktoddbrown/SoilDRaH/issues/3](https://github.com/ktoddbrown/SoilDRaH/issues/3)
    - Is done?
    - Gabriel:
        - narative review looks good
        - More verbose comments
        - fixed a bug in code and pushed changes
        - Create a pull: ISCN3 -> Main
            - DR. TB will perform code review
- NCSS [https://github.com/ktoddbrown/SoilDRaH/issues/5](https://github.com/ktoddbrown/SoilDRaH/issues/5)
    - Report and read function is drafted
        - Still needs some commenting - Gabriel
    - Pull soilDB into its own issue; contact Dylan
    - NCSS uses SQL version (Could impact header names)
- Variable Mapping https://github.com/ktoddbrown/SoilDRaH/issues/6 
    - Dr. TB will export to a variable map
    - What is variable mapping?
        - In merging three different data sets, each set has different vars, metrics, and units
        - Mapping documents transformations/changes required to map source variable to a target variable
    - Dr. TB will prepare a walkthrough of variable mapping for next call
    - Unit Conversions are important for making harmonized data usable
    - Our strategy using value, method, unit tuples
    - Focusing on coding for the purpose of documentation
    - Variable map includes human readable notes to describe important changes ocurring when going from one data product to another
    - Vaasuki proposes using a control vocabulary 
        - Mapping enables easy cross referencing for a merged control vocabularly
        - This is because mapping explicitly describes changes between original and final products
- Geochemical and Mineralogical data https://github.com/ktoddbrown/SoilDRaH/issues/20
    - Source data can not be altered
    - Look for CSV file but use excel if necessary
    - Goal is to restructure for R data annotation in Rmd document
        - We use markdown because we only need to generate annotation once, after which eval=false, then annotations can be stashed as pure documentation
    - Change name to readSmith2019 
- ISRAD https://github.com/ktoddbrown/SoilDRaH/issues/17 
    - Kate needs SoilDRaH push permissions
    - 2024 github version of csv’s are much larger and most likely cleaner
        - Can use newer versions as long as names match our current annotations
        - ISRAD has changed fraction table between old and new versions
        - ^ This is ok as long as we perform checks on format
        - Create an issue for creation of another check
        - Layer level data should be consistent between versions
    - If generation of annotations can be demonstrated on a markdown this is ideal
    - For now just start code up and push; documentation of hand corrections can be done after
    - Dr. TB thinks regex will help with of_type
    - We are moving forward with 2024 version of data and creating a new issue for creating content checking functions
- Longleaf https://github.com/ktoddbrown/SoilDRaH/issues/16
    - Data annotations script can be moved from script to markdown file’s eval=false chunk
        - Switch from get_wide to tidyverse functions
        - Helper functions obscure what’s happening
        - We want to rely on in document functions
        - Coding convention changes
            - Mutate functions in a row can all be in one function call
            - Look at other function for style reference

### (5 | 1450) Soil-DRaH Wiki

- Read Scripts
    - Page under wiki is updated
    - Skeleton of integration workflow
        - Todo item

### (5 | 1455) Check outs [appreciations, curiosities, and ToDo]

- Next meeting will be switched to a morning time slot
- Todo items
    - Dr. TB: Calendar stuff
    - Gabriel: lots of stuff
    - Graham: touch up readme and wiki workflow tab, check with Gabriel regarding code commenting
    - Kate: Pushing ISRAD changes
    - Vaasuki: Looking forward to some reviews
